### PR TITLE
Backport new robotics pkgs

### PR DESCRIPTION
**Pull request by therealpxc**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

Except for the upgrade to Ogre3D (the current version of which no longer builds in 16.03 anyway), all of these commits add new packages I recently submitted to master. 16.03 is the release I actually use them on.

cc @domenkozar 

---

_Please note, that points are not mandatory, but rather desired._

### CODE REVIEW

**Comment by jagajaga** [comment]
We do not backport new pkgs to stable-channels.
We do just pkg updates.
https://github.com/NixOS/nixpkgs/pull/14124#issuecomment-200323737

**Comment by domenkozar** [comment]
Yes we do. I see no harm in doing that. Is there a reason we shouldn't do it?

**Comment by jagajaga** [comment]
@domenkozar We do?

**Comment by jagajaga** [comment]
@domenkozar I supposed that stable channels are for updates and CVE fixes.

**Comment by domenkozar** [comment]
Yes. Stable is there to be, well stable. But new packages provide no harm whatsoever to stability.

**Comment by jagajaga** [comment]
@domenkozar ok.

**Comment by domenkozar** [comment]
We won't commit to adding all new packages, but if someone wants them I see no reason not to backport those packages.

File: pkgs/applications/science/robotics/qgroundcontrol/default.nix [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  # TODO: package mavlink so we can build from a normal source tarball
+  src = fetchgit {
+    url = "https://github.com/mavlink/qgroundcontrol.git";
+    rev = "refs/tags/v${version}";
+    sha256 = "0rwn2ddlar58ydzdykvnab1anr4xzvb9x0sxx5rs037i49f6sqga";
+    fetchSubmodules = true;
+  };
+
+  meta = {
+    description = "provides full ground station support and configuration for the PX4 and APM Flight Stacks";
```
Comment (jagajaga) [inline pkgs/applications/science/robotics/qgroundcontrol/default.nix]: Please start with a capital letter.

**Comment by jagajaga** [comment]
@domenkozar there was my nitpick in a comment to a commit :(

**Comment by domenkozar** [comment]
fixed in 7f2dc5d

**Comment by therealpxc** [comment]
Thanks folks! Sorry for the confusion, and for missing that typo.

**Pull request merged**